src/encoded/tests/test_search.py

```diff
@@ -753,3 +888,115 @@ def test_format_facets_adds_pseudo_facet_for_extra_filters():
         ],
         'total': 42,
     }]
+
+
+def test_sort_query(sort_query_item):
+
+    sorted_query = sort_query(sort_query_item)
+    assert isinstance(sorted_query, OrderedDict) is True
+    for key in sort_query_item:
+        assert sort_query_item[key] == sorted_query[key]
```

Comment:
Did we update this test or is it a copy/paste of an original?

This doesn't test if the keys are now sorted or if the sub dicts are ordered dicts.  The sort_query function does two things.    One it rebuilds the input dict with sorted keys.  Second it calls itself for all sub dicts.  So we should test for these two things.

Also don't use "real" data like sort_query_item() for tests that do not require real data.  It makes everything way more complex than it needs to be.  sort_query doesn't care what kind of dict it gets.

For example, 
```
test_dict = {
    '1': 1,
    '3': 3,
    '2': 2,
}
....
assert list(result_dict.keys()) == sorted(list(test_dict.keys()))
```  
Of course this is a partial example because it doesn't test the recursive property of the function.  The point is to make tests as simple as possible.  I wish there were better examples to follow in our code base but that's one reason we're updating them. 

 I didn't look into the other tests, but try updating them accordingly.  Some of them look data dependent but others don't.

Reply:
Sure. I updated this test with simple inputs and I am validating the output of the function by hardcoding valid OrderedDicts to test the sorting and recursive property of this function. I looked at other tests too. Those are having hardcoded conditions which cannot be tested without partial real data. For example, 

build_terms_filter() has 
if field.endswith('!'):
        field = field[:-1]
        if not field.startswith('audit'):

So, in order to get into these if blocks, I have a partial real data (like 'audit!'). Let me know if this sounds ok.